js/isomer.js

```diff
@@ -67,25 +72,215 @@ Isomer.prototype._translatePoint = function (point) {
 /**
  * Adds a shape or path to the scene
  *
- * This method also accepts arrays
+ * This method also accepts arrays.
+ * By default or if expertMode=0, shapes are drawn
  */
-Isomer.prototype.add = function (item, baseColor) {
+Isomer.prototype.add = function (item, baseColor, expertMode, name) {
+  var expertModeValid = !!expertMode;
   if (Object.prototype.toString.call(item) == '[object Array]') {
     for (var i = 0; i < item.length; i++) {
-      this.add(item[i], baseColor);
+      this.add(item[i], baseColor, expertModeValid, name);
     }
   } else if (item instanceof Path) {
-    this._addPath(item, baseColor);
+    if(expertModeValid){
+      this.paths[this.paths.length] = {path:item, color:baseColor, shapeName:(name||'')};
+    } else {
+      this._addPath(item, baseColor);
+    }
   } else if (item instanceof Shape) {
-    /* Fetch paths ordered by distance to prevent overlaps */
     var paths = item.orderedPaths();
     for (var i in paths) {
-      this._addPath(paths[i], baseColor);
+      if(expertModeValid){
+        this.paths[this.paths.length] = {path:paths[i], color:baseColor, shapeName:(name||'')};
+      } else {
+	this._addPath(paths[i], baseColor);
+      }
+    }
+  }
+};
+
+/**
+ * Draws the content of this.paths
+ * By default, does not sort the paths between shapes
+ */
+Isomer.prototype.draw = function(sortPath){
+  var sortValid = !!sortPath;
+  if(sortValid){
+    this.sortPaths();
+  }
+  for (var i in this.paths){
+    this._addPath(this.paths[i].path, this.paths[i].color);
+  }
+}
+
+
+/**
+ * Sorts the paths contained in this.paths,
+ * ordered so that distant faces are displayed first
+ *  */
+Isomer.prototype.sortPaths = function () {
+  var Point = Isomer.Point;
+  var observer = new Point(-10, -10, 20);
+  var pathList = [];
+  for (var i = 0; i < this.paths.length; i++) {
+    var currentPath = this.paths[i];
+    pathList[i] = {
+      path: currentPath.path,
+      polygon: currentPath.path.points.map(this._translatePoint.bind(this)),
+      color: currentPath.color,
+      drawn: 0,
+      shapeName: currentPath.shapeName
+
+    };
+  }
+ this.paths.length = 0;
+
+ // topological sort
+  
+  var drawBefore = [];
+  for (var i = 0 ; i < pathList.length ; i++){
+	drawBefore[i] = [];
+  }
+  for (var i = 0 ; i < pathList.length ; i++){
+    for (var j = 0 ; j < i ; j++){
+	  if(this._hasIntersection(pathList[i].polygon, pathList[j].polygon)){
```

Comment:
A couple nits here as far as consistent spacing! I think we use two spaces in this library, but here we have 4.